pkg/operator/configobservation/configobservercontroller/observe_config_controller.go

```diff
@@ -87,7 +87,7 @@ func NewConfigObserver(
 				SecretLister_:                kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().Secrets().Lister(),
 				ConfigSecretLister_:          kubeInformersForNamespaces.InformersFor(operatorclient.GlobalUserSpecifiedConfigNamespace).Core().V1().Secrets().Lister(),
 				OpenshiftEtcdEndpointsLister: kubeInformersForNamespaces.InformersFor("openshift-etcd").Core().V1().Endpoints().Lister(),
-				ConfigmapLister:              kubeInformersForNamespaces.InformersFor("openshift-etcd").Core().V1().ConfigMaps().Lister(),
+				ConfigmapLister_:             kubeInformersForNamespaces.ConfigMapLister(),
```

Comment:
won't this be cluster-wide?

Reply:
if you do `List` then you will get all config maps. 

however, our config observers only accept `configobserver.Lister`  which must be cast to the desired type. We cast it to `ConfigMapLister` and use it as follows `ConfigMapLister().ConfigMaps(EtcdEndpointNamespace).Get(etcdEndpointName)` to get scoped resources.